src/bestcrypt_ve_fmt_plug.c

```diff
@@ -0,0 +1,456 @@
+#define BENCHMARK_LENGTH        0x507
+#define BINARY_SIZE             0
+#define PLAINTEXT_LENGTH        125
+#define SALT_SIZE               sizeof(*cur_salt)
+#define BINARY_ALIGN            sizeof(uint32_t)
+#define SALT_ALIGN              sizeof(uint64_t)
+#define MIN_KEYS_PER_CRYPT      1
+#define MAX_KEYS_PER_CRYPT      1
+
+#define OMP_SCALE               1
+
+#define aesId                   8
+#define twofishId               9
+#define serpentId               10
+#define rc6Id                   11
+#define camelliaId              15
+
+#define SERPENT_KS (140 * 4)
+
+static struct fmt_tests tests[] = {
+	{"$bcve$4$8$4944e8bfed6c688b6b30ed3aedb21c9efeba82a1425ebf8e$a4b51d5e6d0b15ba039d0a62cb1a2f4a1d0f6c0325658b15334657b7732a6dc4ec72de7a7c04216d0dd5167bc23e384ac99bca490307445bcbeeaf09da15465ba6540daa52422afdce6f199f92632e0675d9f90693a49ec964beb2358c7a95ff", "openwall"},
+	{"$bcve$4$8$4afeb70b633687b1869fb6f99c988930688391478d730672$e1c4c86980e60aaebb55b2857be43055aae966925812687481e6844a3ded5480f51302cdd633e040a30790abdd8ea520e28ec7a261e4226bdc3019b1905fe1df1fbbbc98b7ee409ef2ba8be359cf18ccc2f9f9dbeb20d75c5445c35d77e03907","B4by$harKtudududu!"},
+	{"$bcve$4$9$06046a1b1bde8dca9df92696f600ce54b0084cdde00e6e07$f708a6916dbe91d105e689089ca70c37dc18e761773154f28063932ead9eebdcf9fb478ce48923408eb15ccfec99e44372d47bff9aca6b96ba10d9f8c0b9d6b57b75f27f359bef6e632370fd701fca2160f46d39fffcefb30e769bafb87cbd32", "openwall123!"},
+	{"$bcve$4$10$65c70a51ec78c46a17e0604804f8ef297b118a42c8109b05$2f92091413f51a446b8576d06790d682b251a97694d19786c45a9245dacb5100fcfc757ba993a83e6f8e13da0d726c57bdf28e70beaa734034c69cd7f655afaebf6c3cf1a0617f673e267b6e5bc7a3619c9eb6801aa8f1a1824ed3cca55ece0d", "!1234openwall"},
+	//{"$bcve$4$11$5fbbb914874770659e46e7f5a72b506076271bfebcaed823$7216a0b623354b0f2ed417a121cbbed39e0a1ce84c86cd4c5dbe75cbe00b0e947f1cda7e10d7ce7c88f479c086aa672d4c4610896cbf81f6dff82e16d0648d3adbadc271058c7f0edb40a599bca7fbc8eefb26be106b182595e4fa68fcf47b9c", "!1234openwall"},
+	{"$bcve$4$15$1386f84fa770e372d9221bb4e4aaf8ee2e0d40e19b69917c$f7a0c8bcdf279fe29f5ab144da729e9dc0c4ca12efb1e2be25e417f30db78227831bb8ee1bee84b0cd37582af2aaa34102fcc851729d676630d7f3de34f6258bd639e6a96301df4038601b30961d35b502506bf4b4d373e9f007b95f9b1abeb2", "openwall123!"},
+	{NULL}
+};
+
+static int max_threads;
+static yescrypt_local_t *local;
+
+static char (*saved_key)[PLAINTEXT_LENGTH + 1];
+static int *saved_len;
+static int *cracked, cracked_count;
+
+static struct custom_salt {
+	int version;
+	int enc_algoID;
+	unsigned char salt[24];
+	unsigned char encrypted_data[0x60];
+} *cur_salt;
+
+typedef Twofish_Byte    Byte;
+typedef Twofish_UInt32  UInt32;
+
+/* ripped from ossl_aes_crypto.c */
+static void serpent_cbc_decrypt(const unsigned char *in, unsigned char *out,
+                                size_t len, void *key,
+                                unsigned char ivec[16])
+{
+	size_t n;
+	union { size_t t[16/sizeof(size_t)]; unsigned char c[16]; } tmp;
```

Comment:
I think `size_t` is irrelevant: internally, `serpent_decrypt` works on `uint32_t`, so that's what we should use here (or at least have as one of the fields in this `union`) to avoid violating C aliasing rules (or at least get close to respecting them). We could also use `ARCH_WORD` or `uint64_t` to speed up our own loops here, but I think that would be trickier and unnecessary.